ccm_web/server/src/api/api.service.ts

```diff
@@ -121,6 +137,46 @@ export class APIService {
     return await sectionHandler.enrollUsers(sectionUsers)
   }
 
+  async enrollSectionExternalUsers (user: User, session: SessionData, sectionId: number, sectionUsers: SectionExternalUserDto[]): Promise<ExternalEnrollments|APIErrorData> {
+    // Get requestor/handler and account ID for admin operations
+    const adminRequestor = this.canvasService.createRequestorForAdmin('/api/v1/')
+    const adminHandler = new AdminApiHandler(adminRequestor, user.loginId)
+    const newUserAccountID =  this.configService.get('canvas.newUserAccountID', { infer: true }) as number
+
+    // Create all requested users, noting failures
+    const createUserResponses = await adminHandler.createExternalUsers(sectionUsers, newUserAccountID)
+    const newUsers = createUserResponses.filter(response => !hasKeys(response, ['error'])) as CanvasUserLoginEmail[]
+
+    console.log('createUserResponses…')
+    console.log(JSON.stringify(createUserResponses))
+
+    // Results of inviting only new users
+    let inviteResults: CirrusInvitationResponse
+    try {
+      inviteResults = await this.invitationService.sendInvitations(newUsers)
```

Comment:
The Cirrus invitation flow is only getting the user list after filtering errors. Which is good thing

Reply:
Yes, that was the original plan for who this code would work.  Even before I implemented it, discussions with Melinda made it clear that only new users should receive Cirrus invitations.

Reply:
🏁